Source/Globals.cpp

```diff
@@ -57,9 +61,47 @@ namespace android::global
     }
 
     void Initialize(JavaVM* javaVM, jobject context)
+    {
+        Initialize(javaVM, context, nullptr);
+    }
+
+    void Initialize(JavaVM* javaVM, jobject context, jobject assetManager)
     {
         g_javaVM = javaVM;
-        g_appContext = GetEnvForCurrentThread()->NewGlobalRef(android::content::Context{context}.getApplicationContext());
+        JNIEnv* env = GetEnvForCurrentThread();
+
+        if (g_appContext != nullptr)
+        {
+            env->DeleteGlobalRef(g_appContext);
+            g_appContext = nullptr;
+        }
+
+        if (context != nullptr)
+        {
+            android::content::Context contextWrapper{context};
+            const auto applicationContextWrapper = contextWrapper.getApplicationContext();
+            jobject applicationContext = applicationContextWrapper;
+
+            if (applicationContext != nullptr)
+            {
+                g_appContext = env->NewGlobalRef(applicationContext);
+            }
+        }
+
+        if (assetManager != nullptr)
+        {
+            SetAssetManager(assetManager);
+        }
+        else if (context != nullptr)
+        {
+            android::content::Context contextWrapper{context};
+            const auto assetsWrapper = contextWrapper.getAssets();
+            SetAssetManager(static_cast<jobject>(assetsWrapper));
```

Comment:
Is this different than what you have?

```suggestion
            SetAssetManager(android::content::Context{context}.getAssets());
```